extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ObjectQueryService.java

```diff
@@ -223,10 +224,11 @@ else if (results.isReferral() && request.isFollowReferrals()) {
                 
             }
 
+            results.close();
```

Comment:
Will this still leak if an exception is thrown above?

Reply:
[`try`-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) might be an option.

Reply:
Okay, gave this another shot with a `try...finally` block.  I don't like the nested `try` statements - I could declare the `results` variable outside of the first `try` block, instead, if you think that would be a better route?

Reply:
Well, I think you can likely avoid all that with Java 8 fanciness (`try`-with-resources). Something like:

```java
// Search within subtree of given base DN
LdapConnectionConfig ldapConnectionConfig = ldapConnection.getConfig();
SearchRequest request = ldapService.getSearchRequest(baseDN, query);

try (SearchCursor results = ldapConnection.search(request)) {
    ...
}
catch (CursorException | IOException | LdapException e) {
    ...
}
```

Reply:
In fact ... I think the same implicit `close()` from `try`-with-resources would work with `LdapNetworkConnection`, as well. It implements `AutoCloseable` just as `SearchCursor` does.

Reply:
Cool - I'll rework it, give me 5 minutes :smile: